File: old/testerum-backend/testerum-runner/testerum-runner-api/src/main/kotlin/com/testerum/runner/events/model/ParametrizedTestStartEvent.kt
package com.testerum.runner.events.model

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.testerum.model.infrastructure.path.Path
import java.time.LocalDateTime

data class ParametrizedTestStartEvent @JsonCreator constructor(
        @JsonProperty("time")         override val time: LocalDateTime = LocalDateTime.now(),
        @JsonProperty("eventKey")     override val eventKey: String,
        @JsonProperty("testName")     val testName: String,
        @JsonProperty("testFilePath") val testFilePath: Path,
        @JsonProperty("tags")         val tags: List<String>
): RunnerEvent

File: old/web/web-frontend/src/app/model/test/event/parametrized-test-start.event.ts
import {RunnerEvent} from "./runner.event";
import {Path} from "../../infrastructure/path/path.model";
import {RunnerEventTypeEnum} from "./enums/runner-event-type.enum";
import {Serializable} from "../../infrastructure/serializable.model";

export class ParametrizedTestStartEvent implements RunnerEvent, Serializable<ParametrizedTestStartEvent> {
    time: Date;
    eventKey: string;
    eventType: RunnerEventTypeEnum = RunnerEventTypeEnum.PARAMETRIZED_TEST_START_EVENT;

    testName: string;
    testFilePath: Path;
    tags: Array<string> = [];

    deserialize(input: Object): ParametrizedTestStartEvent {
        this.eventKey = input["eventKey"];
        this.time = new Date(input["time"]);
        this.testName = input["testName"];
        this.testFilePath = Path.deserialize(input["testFilePath"]);
        this.tags = input['tags'] || [];

        return this;
    }

    serialize(): string {
        return undefined;
    }
}
